ios/BT/ExposureManager.swift

```diff
@@ -2,245 +2,228 @@ import Foundation
-                } else {
-                    completion(["Exposure detection error.", NSNull()])
-                }
-            }
-        case .simulateExposureDetectionError:
-            BTSecureStorage.shared.exposureDetectionErrorLocalizedDescription = "Unable to connect to server."
-            completion([NSNull(), "Exposure deteaction error message: \(BTSecureStorage.shared.exposureDetectionErrorLocalizedDescription)"])
-        case .simulateExposure:
-            let exposure = Exposure(id: UUID().uuidString,
-                                    date: Date().posixRepresentation - Int(TimeInterval.random(in: 0...13)) * 24 * 60 * 60 * 1000,
-                                    duration: TimeInterval(Int.random(in: 1...10) * 60 * 5 * 1000),
-                                    totalRiskScore: .random(in: 1...8),
-                                    transmissionRiskLevel: .random(in: 0...7))
-            let exposures = BTSecureStorage.shared.exposures
-            exposures.append(exposure)
-            BTSecureStorage.shared.exposures = exposures
-            completion([NSNull(), "Exposures: \(BTSecureStorage.shared.exposures)"])
-        case .resetExposureDetectionError:
-            BTSecureStorage.shared.exposureDetectionErrorLocalizedDescription = .default
-            completion([NSNull(), "Exposure Detection Error: "])
-        case .getAndPostDiagnosisKeys:
-            getAndPostTestDiagnosisKeys { error in
-                if let error = error {
-                    completion([error.localizedDescription, NSNull()])
-                } else {
-                    completion([NSNull(), "Local diagnosis keys successfully fetched and posted."])
-                }
-            }
-        case .resetExposures:
-            BTSecureStorage.shared.exposures = List<Exposure>()
-            completion([NSNull(), "Exposures: \(BTSecureStorage.shared.exposures.count)"])
-        case .toggleENAuthorization:
-            let enabled = manager.exposureNotificationEnabled ? false : true
-            requestExposureNotificationAuthorization(enabled: enabled) { result in
-                completion([NSNull(), "EN Enabled: \(self.manager.exposureNotificationEnabled)"])
-            }
+}
+
+private extension ExposureManager {
+
+  @objc func scheduleBackgroundTaskIfNeeded() {
+    guard ENManager.authorizationStatus == .authorized else { return }
+    let taskRequest = BGProcessingTaskRequest(identifier: ExposureManager.backgroundTaskIdentifier)
+    taskRequest.requiresNetworkConnectivity = true
+    do {
+      try BGTaskScheduler.shared.submit(taskRequest)
+    } catch {
+      print("Unable to schedule background task: \(error)")
+    }
+  }
+
+  func notifyUserBlueToothOffIfNeeded() {
+    let identifier = String.bluetoothNotificationIdentifier
+    if ENManager.authorizationStatus == .authorized && manager.exposureNotificationStatus == .bluetoothOff {
```

Comment:
so does this mean that if the user has bluetooth off, but en permissions on we dont have en notifications coming through?  or does turning bluetooth off also turn en permission off automatically?

Reply:
good question - the former; I'll add a comment 👍